tests/test_sending_power_forecast.py

```diff
@@ -0,0 +1,64 @@
+"""
+Test file for the device client. Depends on d3a test setup file strategy_tests.external_devices
+"""
+import logging
+import os
+from time import sleep
+from d3a_api_client.rest_device import RestDeviceClient
+from d3a_api_client.utils import get_area_uuid_from_area_name_and_collaboration_id
+
+root_logger = logging.getLogger()
+root_logger.setLevel(logging.INFO)
+
+
+class AutoSendForecast(RestDeviceClient):
+    forecast = 0
+
+    def on_market_cycle(self, market_info):
+        """
+        Sends increasing energy forecast to pv and load devices
+        """
+        root_logger.debug(f"New market information {market_info}")
+        self.forecast += 50
+        if "available_energy_kWh" in market_info["device_info"]:
+            root_logger.info(f"self.set_pv_power_forecast({self.forecast})")
+            response = self.set_pv_power_forecast(self.forecast)
+
+        if "energy_requirement_kWh" in market_info["device_info"]:
+            root_logger.info(f"self.set_load_power_forecast({self.forecast})")
+            response = self.set_load_power_forecast(self.forecast)
```

Comment:
I have a general comment, this seems like a hacky way, meaning if the strategy contains `available_energy_kWh ` so send the forecasted pv power or if the strategy contains `energy_requirement_kWh ` send the forecasted load power. Im not saying it won't work but it gives a false image to user that while removing
`if "available_energy_kWh" in market_info["device_info"]:` and `if "energy_requirement_kWh" in market_info["device_info"]:` can send the load forecast as pv and the other way around.

TLDR: Better to have separate scripts for pv forecast and load forecast.

Reply:
@faizan2590 Yes, of course, I agree with your comment, but this is really only a script for testing the hardware api. 
I really made it for you. The user (or here: the hardware) should know that it can only send the forecast of a load with `set_load_power_forecast`.